### PR TITLE
Add retain restaurants on reload feat to client

### PR DESCRIPTION
**Pull request by JoshuaRamc**
Added feature that retains the users search query on reload of the page.

### CODE REVIEW

File: client/src/pages/Search/index.js [hunk 1]
```diff
@@ -14,7 +14,10 @@ import useMenuModal from "../../hooks/useMenuModal";
 export default function Search() {
   const dispatch = useDispatch();
   const [restaurants, setRestaurants] = useState([]);
-  const [searchFields, setSearchFields] = useState(DEFAULT_SEARCH_QUERY);
+  const [searchFields, setSearchFields] = useState({
+    ...DEFAULT_SEARCH_QUERY,
+    ...JSON.parse(localStorage.getItem("budget-eats-cache")),
+  });
   const [viewMode, setViewMode] = useState("map");
```
Comment (DLi7077) [inline client/src/pages/Search/index.js:21]: can we also retain the previous view?

**Review by DLi7077 — CHANGES_REQUESTED** [review]

File: client/src/pages/Search/index.js (hunk 1 above)
Comment (JoshuaRamc) [reply client/src/pages/Search/index.js:21]: what do you mean by that?

Comment (JoshuaRamc) [reply client/src/pages/Search/index.js:21]: like if they were on grid or map view, you want us to retain the view they were on?

Comment (DLi7077) [reply client/src/pages/Search/index.js:21]: yeah exactly. Sorry for the late response
Consider if the user prefers grid view.
I go to the restaurant page
I go back - its map view then i gotta go to grid view
rinse repeat, kinda annoying

**Review by DLi7077 — APPROVED** [review]
Lgtm

**Pull request merged**